cmd/zpool/zpool_main.c

```diff
@@ -5174,6 +5181,9 @@ zpool_do_iostat(int argc, char **argv)
 		return (1);
 	}
 
+	if (ioctl(1, TIOCGWINSZ, &win) != -1 && win.ws_row > 0)
```

Comment:
The window size should checked on each iteration.  That will handle a terminal resize while zpool iostat is running.